.editorconfig

```diff
@@ -147,5 +147,139 @@ dotnet_naming_rule.private_field_rule.style = private_member_style
 dotnet_diagnostic.IDE0076.severity = warning
 dotnet_diagnostic.IDE0005.severity = warning
 
-; Code Quality rules: Assumes methods do null checks.
+dotnet_diagnostic.VSTHRD200.severity = none
+
+# NuGet Code Analysis Rules
+# Code files under src/ folder
+[src/**/*.cs]
```

Comment:
I converted the [NuGet.ruleset file](https://github.com/NuGet/NuGet.Client/blob/dev/build/NuGet.ruleset) to an editorconfig file using the [Microsoft.CodeAnalysis.RulesetToEditorconfigConverter](https://www.nuget.org/packages/Microsoft.CodeAnalysis.RulesetToEditorconfigConverter/3.0.0) package.